infrastructure/lambda/redirect/index.mjs

```diff
@@ -0,0 +1,22 @@
+// @ts-check
```

Comment:
This gives *some* type-checking, as opposed to none. For the reasons why this isn't written in typescript, see the PR description footnote.

More on this: https://www.typescriptlang.org/docs/handbook/intro-to-js-ts.html